pkg/cloudinit/common.go

```diff
@@ -101,3 +112,37 @@ func NewBaseCloudConfig(data BaseUserData) (CloudConfig, error) {
 func makeMicroclusterAddress(address string, port int) string {
 	return net.JoinHostPort(address, strconv.Itoa(port))
 }
+
+func getSnapstoreProxyConfigFiles(snapstoreProxyScheme, snapstoreProxyDomain, snapstoreProxyID string) []File {
```

Comment:
You conditionally write the snapstore config files (scheme, domain, id files) but always write and execute the `configure-snapstore-config.sh`. 
In the script, you fail/exit if any of the configs are not set. 

This breaks cloud-init in the default configuration.

Reply:
It doesn't break, because cloud-init doesn't stop when one of it's runcmds fail. Regardless, I'll make this more robust.